### PR TITLE
feat(ui) Add echarts based stacked bar chart

### PR DESCRIPTION
**Pull request by markstory**
Add an echarts based wrapper for replacing the StackedBarChart
component. This will help normalize tooltips and colours used across
charts. It also fixes some rendering issues with StackedBarChart where
a small empty block is shown in all charts.

I've not integrated the new component into any views yet as I wanted to
get feedback on the design and API for this new component. Eventually
I'd like to replace all of the StackedBarChart usage across the
application with this component.

barchart

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+  /**
+   * Colors to use on the chart.
+   */
+  colors: string[];
+
+  /**
+   * Hover state colors to use on the chart.
+   */
+  emphasisColors: string[];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:40]:
```suggestion
  /**
   * Colors to use on the chart.
   */
  colors: string[];
  /**
   * Hover state colors to use on the chart.
   */
  emphasisColors: string[];
    /**
   * A list of series to be rendered as markLine components on the chart
   * This is often used to indicate start/end markers on the xAxis
   */
  markers?: Marker[];
  /**
   * Whether timestamps are should be shown in UTC or local timezone.
   */
  utc?: boolean;
```

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+};
+
+class MiniBarChart extends React.Component<Props> {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:43]: maybe we should convert it to a function?

File: docs-ui/components/barChart.stories.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import {withInfo} from '@storybook/addon-info';
 
+import MiniBarChart from 'app/components/charts/miniBarChart';
```
Comment (priscilawebdev) [inline docs-ui/components/barChart.stories.js:4]:
```suggestion
```

Comment (priscilawebdev) [reply docs-ui/components/barChart.stories.js:4]: it looks like this is not being used

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx (hunk 1 above)
Comment (matejminar) [inline src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:40]: Should we move `colors` and `emphasisColors` to own `DefaultProps` type? Right now we can pass whatever we want into `defaultProps` object and the compiler would not complain.

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  color: string;
+  symbolSize?: number;
+};
+
+const defaultProps = {
+  colors: [theme.gray300, theme.purple300, theme.purple400],
+  emphasisColors: [theme.gray400, theme.purple400, theme.purple500],
+};
+
+type Props = React.ComponentProps<typeof BarChart> & {
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:21]: good catch @matejminar !!

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/charts/miniBarChart.tsx]: Sorry I misread this. I've added better types for defaultProps.

File: src/sentry/static/sentry/app/components/charts/miniBarChart.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+type Props = React.ComponentProps<typeof BaseChart> &
+  DefaultProps & {
+    /**
+     * Colors to use on the chart.
+     */
+    colors: string[];
+    /**
+     * Hover state colors to use on the chart.
+     */
+    emphasisColors: string[];
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/charts/miniBarChart.tsx:43]: We can remove these now that they are in DefaultProps

**Review by matejminar — APPROVED** [review]

**Pull request merged**